.github/workflows/publish-docs.yml

```diff
@@ -1,28 +1,52 @@
 name: Publish Docs in GitHub Pages
+
 on:
+  workflow_dispatch: # allow manual runs
   push:
     branches:
       - main
+    paths: # avoid extra builds
+      - docs/**
+      - mkdocs.yml
+      - .github/workflows/publish-docs.yml
+
 permissions:
-  contents: write
+  contents: read
+  pages: write
+  id-token: write
```

Comment:
@janbrasna can you provide some context as to why is this needed?

Reply:
Ah yea, that's the case: [Deploying GitHub Pages artifacts](https://docs.github.com/en/pages/getting-started-with-github-pages/using-custom-workflows-with-github-pages#deploying-github-pages-artifacts)

> The deploy-pages action handles the necessary setup for deploying artifacts. To ensure proper functionality, the following requirements should be met:
> 
> The job must have a minimum of pages: write and id-token: write permissions.